File: benchmarks/tsb/bench_quantile.ts
import { quantile } from "tsb";
const N = 100_000;
const sorted = Array.from({ length: N }, (_, i) => i * 0.001);
const WARMUP = 3;
const ITERS = 20;
for (let i = 0; i < WARMUP; i++) {
  quantile(sorted, 0.25);
  quantile(sorted, 0.5);
  quantile(sorted, 0.75);
}
const t0 = performance.now();
for (let i = 0; i < ITERS; i++) {
  quantile(sorted, 0.25);
  quantile(sorted, 0.5);
  quantile(sorted, 0.75);
}
const total = performance.now() - t0;
console.log(JSON.stringify({ function: "quantile", mean_ms: total / ITERS, iterations: ITERS, total_ms: total }));

File: benchmarks/pandas/bench_quantile.py
import pandas as pd, time, json
import numpy as np
N = 100_000
data = [i * 0.001 for i in range(N)]
s = pd.Series(data)
WARMUP = 3
ITERS = 20
for _ in range(WARMUP):
    s.quantile(0.25)
    s.quantile(0.5)
    s.quantile(0.75)
t0 = time.perf_counter()
for _ in range(ITERS):
    s.quantile(0.25)
    s.quantile(0.5)
    s.quantile(0.75)
total = (time.perf_counter() - t0) * 1000
print(json.dumps({"function": "quantile", "mean_ms": total / ITERS, "iterations": ITERS, "total_ms": total}))
